src/base/memory_monitor_sr.cpp

```diff
@@ -51,3 +90,20 @@ namespace cubmem
     return filecopy + ':' + std::to_string (line);
   }
 }
+
+using namespace cubmem;
+
+bool mmon_is_mem_tracked ()
+{
+  return (mmon_Gl != nullptr);
+}
+
+size_t mmon_get_alloc_size (char *ptr)
+{
+  if (mmon_is_mem_tracked ())
+    {
+      return mmon_Gl->get_alloc_size (ptr);
+    }
+  // unreachable
+  return 0;
```

Comment:
If this is unreachable
```suggestion
  assert (mmon_is_mem_tracked ()); // unreachable
  
  return mmon_Gl->get_alloc_size (ptr);
```

Reply:
`assert` is not called in the release build, following the approach you suggested may lead to a SIGSEGV. Is this intentional, given that it is unreachable?

Reply:
OK. What happens it returns 0? what does it mean? How is is used?

Reply:
This function is called only when the 'memory_monitoring' option is enabled in cub_realloc(), so this function cannot be called in situations where the memory_monitor is NULL. Therefore, if this function returns 0, it indicates a serious bug. While I initially considered handling this as an error, if your suggestion implies that a SIGSEGV should occur for such a serious bug, I believe a more intuitive approach is needed. The action of calling `return mmon_Gl->get_allocated_size()` is not intuitive because it would cause a SIGSEGV, but it requires understanding the context of the function's operation. I agree with adding an assert.

Reply:
I see. I agree that this feature should not cause a SIGSEGV on an error. To achieve this, we should handle when it returns 0 by the caller. Let's see when the caller code comes in :).